Intersect.Client/Entities/Events/Event.cs

```diff
@@ -77,6 +77,11 @@ public override bool Update()
 
         protected bool TryEnsureTexture(out GameTexture texture)
         {
+            if (Graphic != null && !string.IsNullOrEmpty(Graphic.Filename) && Sprite != Graphic.Filename)
```

Comment:
Remove `Sprite != Graphic.Filename` -- the setter for `Sprite` already makes this check.